### PR TITLE
minimize initialization

### PR DESCRIPTION
**Pull request by oculus-ruben**
Update so that we don't have to compute the `moveMap` for areas we may not care about.  Not sure whether this is the best implementation, so let me know what you think!

I also included some pybind warning fix (feel free to remove)

### CODE REVIEW

**Comment by oculus-ruben** [comment]
Updated per feedback.  I think this is roughly in line with what you were mentioning.  Note: I wasn't sure whether to include the `_` prefix in the relevant functions - feel free to change or I can!

File: House3D/house.py [hunk 1]
```diff
@@ -577,14 +576,42 @@ def is_door(obj):
...
 
+
+    def _getRegionsOfInterest(self):
+        """Override this function for customizing the areas of the map to
+        consider when marking valid movable locations
+        Returns a list of (x1, y1, x2, y2) tuples representing bounding boxes
+        of valid areas.  Coordinates are normalized grid coordinates.
+        """
+        result = []
+        if restrictToTargets:
```
Comment (ppwwyyxx) [inline House3D/house.py]: This variable was declared nowhere? I guess this branch can just be removed?

Comment (oculus-ruben) [reply House3D/house.py]: Oh weird, I think I forgot to include some change to this file - this function should only have this line:
```
return [(0, 0, self.n_row+1, self.n_row+1)]
```

And then the comment is valid for having overridable functionality here

File: House3D/house.py [hunk 2]
```diff
@@ -577,14 +576,31 @@ def is_door(obj):
...
                         self.moveMap[i,j] = 1
 
+
+    def _getRegionsOfInterest(self):
+        """Override this function for customizing the areas of the map to
+        consider when marking valid movable locations
+        Returns a list of (x1, y1, x2, y2) tuples representing bounding boxes
+        of valid areas.  Coordinates are normalized grid coordinates.
+        """
+        return [(0, 0, self.n_row+1, self.n_row+1)]
```
Comment (oculus-ruben) [inline House3D/house.py]: Okay, I think this is correct now

**Pull request merged**